rust/arrow/src/array/transform/mod.rs

```diff
@@ -687,4 +728,118 @@ mod tests {
         let expected = Int16Array::from(vec![Some(1), None]);
         assert_eq!(result.keys(), &expected);
     }
+
+    #[test]
+    fn test_struct() {
+        let strings: ArrayRef = Arc::new(StringArray::from(vec![
+            Some("joe"),
+            None,
+            None,
+            Some("mark"),
+            Some("doe"),
+        ]));
+        let ints: ArrayRef = Arc::new(Int32Array::from(vec![
+            Some(1),
+            Some(2),
+            Some(3),
+            Some(4),
+            Some(5),
+        ]));
+
+        let array =
+            StructArray::try_from(vec![("f1", strings.clone()), ("f2", ints.clone())])
+                .unwrap()
+                .data();
+        let arrays = vec![array.as_ref()];
+        let mut mutable = MutableArrayData::new(arrays, false, 0);
+
+        mutable.extend(0, 1, 3);
```

Comment:
I wonder if ensuring the slice covers an actual string value would be useful (this slice just takes the two `None`, `None` values). It it it was like `mutable.extend(0, 2, 4)` that would also include `Some("mark")`. The same comment applies to `test_struct_nulls` as well

Reply:
I am sorry, I did not notice this comment. You are right, I will address it on a separate PR.